### PR TITLE
JBEAP-1388: Minor issues in jaxws-retail quickstart

### PR DESCRIPTION
**Pull request by jbliznak**
proposed changes

### CODE REVIEW

**Comment by sgilda** [comment]
@jbliznak : Would you mind fetching the latest upstream, rebasing, and squashing to make this easier to test and review? Thanks!

**Comment by sgilda** [comment]
I tried to rebase this and and fix the the conficts, the build but had no luck. I'll leave this one to you.

**Comment by jbliznak** [comment]
@sgilda, squashed and rebased

**Comment by sgilda** [comment]
@jbliznak : When I test this change, even before I rebase to get the latest upstream changes, I get the following build error on the 'mvn clean install wildfly:deploy':

[ERROR] Failed to execute goal on project jboss-jaxws-retail-service: Could not resolve dependencies for project org.jboss.quickstarts.eap:jboss-jaxws-retail-service:war:7.0.0-SNAPSHOT: The following artifacts could not be resolved: org.apache.cxf:cxf-rt-bindings-soap:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-bindings-object:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-bindings-coloc:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-frontend-jaxws:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-transports-http:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-transports-http-hc:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-transports-jms:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-transports-local:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-ws-mex:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-ws-rm:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-ws-policy:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-rt-ws-security:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-tools-java2ws:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-tools-wsdlto-core:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-tools-wsdlto-databinding-jaxb:jar:3.1.2-jbossorg-2, org.apache.cxf:cxf-tools-wsdlto-frontend-jaxws:jar:3.1.2-jbossorg-2, org.apache.cxf.services.ws-discovery:cxf-services-ws-discovery-api:jar:3.1.2-jbossorg-2: Failure to find org.apache.cxf:cxf-rt-bindings-soap:jar:3.1.2-jbossorg-2 in http://jboss-developer.github.io/temp-maven-repo/ was cached in the local repository, resolution will not be reattempted until the update interval of jboss-developer-staging-repository has elapsed or updates are forced -> [Help 1]

Does this build for you?

**Comment by jbliznak** [comment]
Well, I don't get these errors, because I have it available in local repository...
This artifact is available in maven repository zip coming with EAP and also in http://repository.jboss.org/nexus/content/groups/public/org/apache/cxf/cxf-core/3.1.2-jbossorg-2/. I don't know what the policy is about used repositories.

**Comment by sgilda** [comment]
@jbliznak : Sorry, but the quickstart has to run in the Maven repository or repository configured in the quickstart POM file.

**Comment by jbliznak** [comment]
So I can configure http://repository.jboss.org/nexus/content/groups/public/ repository in this quickstart's POM? I pushed an updated POM with this repository added

**Comment by sgilda** [comment]
@jbliznak : My understanding is that these quickstarts must work with the EAP repositories and can not depend on Nexus, but I will need to confirm this with @rafabene and @pgier .

**Comment by pgier** [comment]
@jbliznak Will the quickstart still work if you downgrade cxf to the upstream release (3.1.2)?  We'd like to avoid depending on the JBoss.org repository if possible.

**Comment by jbliznak** [comment]
@pgier It will work, but I think the exclusion of all cxf dependencies from jbossws-cxf-client and readding them to quickstart would make the quickstart harder to read and definitely not showing the ease of use.

The other options here are:
1) get these jbossorg artifacts to EAP repository (for example EAP Alpha will most certainly use them, so it might resolve this)
2) wait for next jbossws-cxf release which will use only upstream dependencies
3) use beta version of jbossws client, as it use only public dependencies http://mvnrepository.com/artifact/org.jboss.ws.cxf/jbossws-cxf-client/5.1.0.Beta1

Choose which one suits you most and I will push updated files.

File: jaxws-retail/pom.xml [hunk 1]
```diff
@@ -128,7 +148,7 @@
          any version of JBoss EAP that implements Java EE. -->
          <dependency>
             <groupId>org.jboss.bom</groupId>
-            <artifactId>jboss-eap-javaee7-with-tools</artifactId>
+            <artifactId>jboss-javaee7-eap</artifactId>
```
Comment (pgier) [inline jaxws-retail/pom.xml:151]: This one should be "jboss-eap-javaee7" for the artifactId.

**Comment by pgier** [comment]
@jbliznak I discussed with sgilda and I think option 3 (use the beta version) would be best for now.  After alpha this won't really be an issue anymore because we will have a separate product branch which will use the productized (-redhat) versions of everything.  Can you update the PR one more time?

Also do you think jbossws-cxf-client should be version managed in the BOM, seems like this qualifies as public API.

**Comment by sgilda** [comment]
@jbliznak : 

I modified your pull to remove the Nexus repo, update the version to 5.1.0.Beta1, and fix the artifactId that Paul mentioned. I tested it and it works great. I tried to issue a pull of my commit back to you, but was not able to find you in the list. This was my commit: https://github.com/sgilda/jboss-eap-quickstarts/commit/1aaee1bb744856755bccbe5252235f0c78611f76

I can add and squash my commit with yours or you can make the changes. Up to you. :-)

**Comment by sgilda** [comment]
I applied the fixes to your commit and merged in pull 1658

Thanks @jbliznak !

**Pull request closed without merging**